src/helpers.cpp

```diff
@@ -246,6 +246,29 @@ namespace ada::helpers {
     if (url.query.has_value()) return;
     while (!url.path.empty() && url.path.back() == ' ') { url.path.resize(url.path.size()-1); }
   }
+
+  ada_really_inline size_t find_authority_delimiter(const bool& is_special, std::string_view view) noexcept {
+    size_t location = 0;
+    bool location_found{false};
+
+    if (is_special) {
+      for (; location < view.size(); ++location) {
+        if (view[location] == '@' || view[location] == '/' || view[location] == '?' || view[location] == '\\') {
+          location_found = true;
+          break;
```

Comment:
You know, instead of break, you could just do 'return location'.

Reply:
Somehow, the performance was worse. I think compiler is optimizing this and an early return inhibits that.

Reply:
Yes, I also tested with the early return and the performance gets considerably worse

Reply:
Also I had a **for** version of the code I wrote, and the **for** version had a worse performance.. but I guess the trick here is the 2 fors instead of just 1 (as I did)

Reply:
Possible, yeah, but the boolean variable is probably not helping.